### ISSUE TITLE
Linter incorrectly marks the #[allow()] macro as being unused variable/unused mut

### ISSUE THREAD

**Issue by tsj845** [rust-lang/rust-analyzer#21959]
**rust-analyzer version**: rust-analyzer version: 0.3.2845-standalone (f1297b2111 2026-03-29)

**rustc version**: rustc 1.83.0 (90b35a623 2024-11-26)

**editor or extension**: VSCode

**relevant settings**: N/A

**repository link (if public, optional)**: N/A

**code snippet to reproduce**:
```rust
// tokio = { version = "^1.50.0", features = ["full"] }

use tokio_tungstenite::connect_async; // tokio-tungstenite = "^0.29.0"
use futures_util::{SinkExt, StreamExt}; // futures-util = { version = "^0.3.32", default-features = false, features = ["std"]}


let (ws_stream, _) = connect_async("some URI").await?;
#[allow(unused_variables,unused_mut)]
let (mut ws_write, mut ws_read) = ws_stream.split();
let _: tokio::task::JoinHandle<Result<(), Box<dyn Error + Send>>> = tokio::spawn(async move {
    while let Some(msg) = ws_read.next().await {
        // do things with msg
    }
    return Ok(());
});

```
<img width="958" height="468" alt="Image" src="https://github.com/user-attachments/assets/d1739b92-21e9-4347-aebf-24726b17efa0" />

Addendum: this issue persisted after restarting the rust analyzer extension.
Addendum: this issue persisted after reinstalling the extension.

**Comment by tsj845** [comment]
Update, the issue persists even when the lines are commented.

<img width="479" height="242" alt="Image" src="https://github.com/user-attachments/assets/6e866133-e94f-4579-8b48-f3bec9f17692" />

**Comment by tsj845** [comment]
I was finally able to get the linter to stop marking random lines with those warnings long after I deleted the relevant lines by deleting my target directory. The erroneous warnings have not returned even after reverting to the code that caused them.

**Comment by ChayimFriedman2** [comment]
This is likely a Cargo or rustc issue then.
